src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -44,19 +48,42 @@ public void check(JavaClass javaClass, ConditionEvents events) {
     }
   }
 
+  @SneakyThrows
+  private Set<Class<?>> extractAllClassesReferencedFromPluginXmlAttributes() {
+    val classLoader = Thread.currentThread().getContextClassLoader();
+    val db = DocumentBuilderFactory.newInstance().newDocumentBuilder();;
+    val doc = db.parse(classLoader.getResourceAsStream("META-INF/plugin.xml"));
+    val nodeList = doc.getElementsByTagName("*");
+    Set<Class<?>> result = new HashSet<>();
+    for (int i = 0; i < nodeList.getLength(); i++) {
+      val node = nodeList.item(i);
+      val attributes = node.getAttributes();
+      for (int j = 0; j < attributes.getLength(); j++) {
+        val attribute = attributes.item(j);
+        val maybeFqcn = attribute.getNodeValue();
+        try {
+          val clazz = Class.forName(maybeFqcn, /* initialize */ false, classLoader);
+          result.add(clazz);
+        } catch (ClassNotFoundException e) {
+          /* ignore */
```

Comment:
don't we need any handling of this case? or is it handled in some other way? if so it may be worth to leave some comment here why we ignore it

Reply:
Inlined the constant completely